src/components/Initialization/Initialization.js

```diff
@@ -38,27 +39,82 @@ class Initialization extends PureComponent<Props, State> {
     logger.logEvent('load-application', {
       node_version: nodeVersion,
     });
-    window.addEventListener('beforeunload', this.killAllRunningProcesses);
-  }
 
-  componentWillUnmount() {
-    window.removeEventListener('beforeunload', this.killAllRunningProcesses);
+    ipcRenderer.on('app-will-close', this.appWillClose);
   }
 
-  killAllRunningProcesses = () => {
+  appWillClose = () => {
+    const { isQueueEmpty, queue, projects } = this.props;
+
+    const activeActions = Object.keys(queue).map(projectId => ({
+      name: projects.find(project => project.id === projectId).name,
+      pending: queue[projectId],
+    }));
+
+    const actionCaption = {
+      install: 'Installing',
+      uninstall: 'Uninstalling',
+    };
+
+    // Map actions to the following string format (multiple projects & multiple queued tasks). For each project it will be a string like:
+    // Task in project <project.name>:\n
+    // - <Installing or Queued> <count> task(s)\n
+    const mapActionsToString = activeActions
+      .map(actionItem =>
+        [
+          `Tasks in project ${actionItem.name}:`,
+          actionItem.pending
+            .map(
+              task =>
+                '* ' +

```

Comment:
I think `-` reads better than `*` in plain text

Reply:
Yes, you're right. I'll change this later today. 
Do you know if it's possible to make text bold in a dialog? I'd like to make the project name bold.

Reply:
I don't think so no